osu.Game/Utils/HumanizerUtils.cs

```diff
@@ -26,5 +26,32 @@ public static string Humanize(DateTimeOffset input)
                 return input.Humanize(culture: new CultureInfo("en-US"));
             }
         }
+
+        /// <summary>
+        /// Turns the current or provided big number into a readable string
+        /// </summary>
+        /// <param name="input">The number to be humanized</param>
+        /// <returns>simplified number with a suffix</returns>
+        public static string Humanize(long input)
+        {
+            int k = 1000;
+
+            var suffixes = new[]
+            {
+                "",
+                "k",
+                "million",
+                "billion",
+                "trillion",
+                "quadrillion",
+                "quintillion",
+            };
+
+            if (input < k)
+                return input.ToString();
+
+            int i = (int)Math.Floor(Math.Log(input) / Math.Log(k));
```

Comment:
Mind covering this with edge case unit tests (10<sup>n - 1</sup> and 10<sup>n</sup>) just for safety's sake? It's not immediately obvious to me this will behave correctly due to taking the natural logarithm (base *e*) instead of base 10. It's *mathematically* sound, but floats do float things.

Also, you might as well take `Math.Log(input, k)`, no?

Reply:
Yeah, you are right about 10(3n - 1)